src/main/webapp/app/entities/feedback.model.ts

```diff
@@ -114,6 +150,40 @@ export class Feedback implements BaseEntity {
         return Feedback.hasDetailText(that) || !!(that.gradingInstruction && that.gradingInstruction.feedback);
     }
 
+    /**
+     * Get the referenced file path for referenced programming feedbacks, or undefined.
+     * Typical reference format for programming feedback: `file:src/com/example/package/MyClass.java_line:13`.
+     * Example output in this case: `src/com/example/package/MyClass.java`
+     */
+    public static getReferenceFilePath(feedback: Feedback): string | undefined {
+        if (!feedback.reference?.startsWith(this.programmingReferencePrefix)) {
+            // Find "file:" prefix
+            // No programming feedback
+            return undefined;
+        }
+        const indexOfLine = feedback.reference?.lastIndexOf(this.programmingReferenceLineSeparator);
+        return feedback.reference.substring(this.programmingReferencePrefix.length, indexOfLine); // Split after "_line:"
+    }
+
+    /**
+     * Get the referenced line for referenced programming feedbacks, or undefined.
+     * Typical reference format for programming feedback: `file:src/com/example/package/MyClass.java_line:13`.
+     * Example output in this case: 13
+     */
+    public static getReferenceLine(feedback: Feedback): number | undefined {
+        if (!feedback?.reference?.startsWith(this.programmingReferencePrefix)) {
+            // Find "file:" prefix
+            // No programming feedback
+            return undefined;
+        }
+        const indexOfLine = feedback.reference?.lastIndexOf(this.programmingReferenceLineSeparator); // Split before "_line:"
+        const line = parseInt(feedback.reference?.substring(indexOfLine + this.programmingReferenceLineSeparator.length));
```

Comment:
Why does `reference` needs a present-check here, but not in getReferenceFilePath?

Reply:
It doesn't need it: The `if` check before would always `return undefined` before that. I've removed the check.